src/dmd/statementsem.d

```diff
@@ -1899,10 +1886,10 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
         ec = new VarExp(Loc.initial, fdapply, false);
         ec = new CallExp(loc, ec, fs.aggr, fexp);
         ec.type = Type.tint32; // don't run semantic() on ec
-        return ec;
+        return cast(CallExp)ec;
```

Comment:
Or you could create a new variable for returning, following the unwritten "set once" rule for temporaries.

Reply:
Its not  temporary, see the line above.

Reply:
I meant locals in general.